### PR TITLE
WINDUP-1221: Flip the default to offline mode and add an --online switch

### PR DESCRIPTION
**Pull request by jsight**
Related: https://github.com/windup/windup-rulesets/pull/179

### CODE REVIEW

**Comment by mareknovotny** [comment]
All looks fine, but could we have both options? just for backward compatibility? just an idea, I know we have increasing the available options over a wanted numbers

**Comment by jsight** [comment]
We could have the other one there as a noop. Having said that, if you specify it now, nothing bad happens, it just gets silently ignored. :)

File: reporting/impl/src/main/resources/reports/templates/include/navbar_macro.ftl [hunk 1]
```diff
@@ -42,11 +42,15 @@
...
+<#macro renderNavbarJavaScript index_page=false>
     <script type="text/javascript">
         var script   = document.createElement("script");
         script.type  = "text/javascript";
-        script.src   = "resources/js/navbar.js";
+        <#if index_page>
+            script.src   = "reports/resources/js/navbar.js";
+        <#else>
+            script.src   = "resources/js/navbar.js";
+        </#if>
```
Comment (OndraZizka) [inline reporting/impl/src/main/resources/reports/templates/include/navbar_macro.ftl]: I guess this is not related to --online ?

**Comment by OndraZizka** [comment]
Compilation failure
[ERROR] /var/lib/jenkins/workspace/windup-core-PR-dependents-check/windup-rulesets/src/test/java/org/jboss/windup/rules/tests/WindupRulesTest.java:[324,28] cannot find symbol
[ERROR] symbol:   method setOffline(boolean)
[ERROR] location: variable windupConfiguration of type org.jboss.windup.exec.configuration.WindupConfiguration
[ERROR] -> [Help 1]

**Comment by OndraZizka** [comment]
retest
